app/main.py

```diff
@@ -1 +1,45 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        if skill not in self.skills:
+            self.skills.extend([skill])
```

Comment:
An interesting decision, it's really logical, that the skills list shouldn't contain duplicates, but that of course hasn't been written in the task description.
But if it was, the great solution would be not to check for existence each time, but to simply use `set` instead of `list`